bibloteket.py

```diff
@@ -0,0 +1,178 @@
+#Huvudmeny
+def bok():
+    böcker = open("bok.txt", "a")
+    a = 1
+#Funktionen börjar med en while loop 
+    while a == 1:
+        try:
+            titel = input("Titel: ")
+            författare = input("författare: ")
+            sidor = int(input("antal sidor: "))
+            inköpsår = int(input("inköpsår: "))    
+            inköpspris = float(input("inköpspris:"))
+            t = 2020 - inköpsår
+#Kooden går vidare till variabler 
+            if t < 50:
+                värde = inköpspris * 0.9 ** t
+            else:
+                värde = inköpspris * 1.08 ** t
+#här sparar vi alla variabler till text filen
+            böcker.write((titel) + 'Författare:  ' + (författare)+ 'Antal sidor:  ' +str(sidor) + 'Inköpsåt:  ' +str(inköpsår)+ 'Inköpspris:  ' +str(inköpspris)+ 'Nuvarande värde:  '+ str(värde)+ '\n')
+            böcker.close()
+            a+=1
+        except ValueError:
+            print("Ogiltigt inmat, prova igen.")
+            continue
+#Musik funktionen är uppbyggd på exakt samma sätt som registreringen av böcker, med undantag att beräkningen av det nuvarande värdet är lite annorlunda
+def cd():
+    musik = open("cd.txt", "r+")
+    a=1
+    while a == 1:
+        try:
+            titel = input("Titel: ")
+            artist = input("Artist: ")
+            låtar = input("antal låtar: ")
+            inköpsår = int(input("inköpsår: "))
+            inköpspris = int(input("inköpspris:"))
+            rad = 0
+            skivor = ["a"]
+                
+            for line in musik:
+                rad += 1
+                if titel in line:
+                    skivor.append((rad, line.rstrip()))
+            antal = len(skivor)
+            värde = round(inköpspris/antal)
+            musik.write((titel)+'   Artist:  ' + (artist)+'  Antal låtar:  ' +str(låtar)+'  Inköpsår:  ' +str(inköpsår)+'  Inköpspris:   ' +str(inköpspris)+ '  Nuvarande värde:  '+str(värde)+"\n")
+            a += 1  
+
+
+        except ValueError:
+            print("Ogiltigt inmat, prova igen")
+            continue
+
+
+def film():
+    filmer = open("film.txt", "a")
+    a =1
+    while a == 1:
+        try:
+            titel = input("Titel: ")
+            regisör = input("Regisör: ")
+            tid = input("antal minuter lång: ")
+            inköpsår = int(input("inköpsår: "))
+            inköpspris = float(input("inköpspris:"))
+            skick = int(input("Ange filmens skick 1-10, 10 är nyskick."))
+            t = 2020 - inköpsår
+            värdet = inköpspris*0.9**t
+            a = skick / 10
+            värde = värdet * a
+            filmer.write((titel)+'Reg: ' + (regisör)+'Tid i min:  ' +str(tid)+'INköpsår:  ' +str(inköpsår)+'Inköpspris:  ' +str(inköpspris)+'Nuvarande värde:  '+ str(värde)+ "\n")
+            a+=1
+        except ValueError:
+            print("ogiltigt svar, prova igen.")

```

Comment:
Finns det en anledning till att man får ett anat felmeddelande här än när man matar in CD eller bok?

Reply:
jag tänkte så här om allt detta fungerar adderar vi 1 till a för att avsluta funktionen och återgå till main funktionen, men om ett ogiltigt input sker nånstans i funktionen använder vi oss av except valueerror och skickar tillbaka användaren till början av funktionen.